### PR TITLE
Integrate with Hugging Face, add `predict_entities` helper method

### PR DESCRIPTION
**Pull request by tomaarsen**
Hello!

## Preface
I'm Tom, a Machine Learning Engineer at Hugging Face, though before I started working there, I was very much into Named Entity Recognition. I developed a [library](https://github.com/tomaarsen/SpanMarkerNER) for NER, which has since seen a nice amount of usage. I'm quite fascinated by the performance that you reach with GLiNER - it was my belief previously that zero-shot NER would not be viable with BERT-like models, and it seems that you've proven me wrong. Great work here!

Now, beyond being a big fan of NER, I'm also a big fan of ease of use & reproducibility. I'm excited to see that you've uploaded two pretrained models & a great demo, and I wanted to help out with making the models even easier to use. So, I've created this pull request which integrates with Hugging Face by roughly following [this integration guide](https://huggingface.co/docs/huggingface_hub/en/guides/integrations).

## Pull Request overview
* Make `GLiNER` a subclass of `PyTorchModelHubMixin`, which adds `from_pretrained` & `push_to_hub` functionality.
* Implement a custom `save_pretrained` method
  * This method saves the model as `pytorch_model.bin` and the config as `gliner_config.json`.
* Implement a custom `_from_pretrained` method
* Add `predict_entities` helper method
* Add model cards for GLiNER-base and GLiNER-multi (outside of this PR, on Hugging Face)

## Implement a custom `_from_pretrained` method
This method is used internally in `from_pretrained` (which is what users are recommended to use). In particular, it:
1. Supports models saved as "gliner_base.pt" & "gliner_multi.pt" (i.e. what you currently have). I think this format can be a little bit confusing, as it's hard to see what the configuration options are (e.g. the `max_len` or the base `model_name`), but I've supported it for compatibility. You can use it like:
```python
from model import GLiNER

model = GLiNER.from_pretrained("urchade/gliner_base")
# or
model = GLiNER.from_pretrained("urchade/gliner_multi")

# or with local files:

model = GLiNER.from_pretrained("local_gliner_base")
# or
model = GLiNER.from_pretrained("local_gliner_multi")
# where `local_gliner_base` and `local_gliner_multi` are local directories with `gliner_base.pt` and `gliner_multi.pt` in them, respectively
```

2. Additionally, it supports a new format that I've introduced in this PR: with `pytorch_model.bin` and `gliner_config.json`. This is just like what you've done, but it saves the `state_dict` in `pytorch_model.bin` and the config in `gliner_config.json`, as opposed to everything in one file. Loading a model with this format is just as simple:
```python
from model import GLiNER

model = GLiNER.from_pretrained("tomaarsen/gliner_base")
# or
model = GLiNER.from_pretrained("tomaarsen/gliner_multi")

# or with local files:

model = GLiNER.from_pretrained("local_gliner_base")
# or
model = GLiNER.from_pretrained("local_gliner_multi")
# where `local_gliner_base` and `local_gliner_multi` are local directories with `pytorch_model.bin` and `gliner_config.json` in them
```

## A new `save_pretrained` method
This new method allows you to save a model in the format that I've described above. Obviously, I don't mean to impose a new format on you, so feel free to modify this to just use the original saving function like before, if you prefer. You can use it like so:

```python
from model import GLiNER

model = GLiNER.from_pretrained("urchade/gliner_base")
model.save_pretrained("local_gliner_base")
# and it'll save it to a directory called 'local_gliner_base'

# or
model.push_to_hub("tomaarsen/my_gliner_model")
# and it'll push the model to the Hugging Face Hub instead, under that repository name
```

## Add `predict_entities` helper method
I've introduced a helper function based on your demo code. It returns a list of entities, where each entity is a dictionary with:
* `start`: The start index that can be used to slice the input text.
* `end`: The end index that can be used to slice the input text.
* `text`: The actual entity that's being recognized.
* `label`: The corresponding label.

So, we can now simplify your README snippet to:
```python
from model import GLiNER

model = GLiNER.from_pretrained("urchade/gliner_base")

text = """
Cristiano Ronaldo dos Santos Aveiro (Portuguese pronunciation: [kɾiʃˈtjɐnu ʁɔˈnaldu]; born 5 February 1985) is a Portuguese professional footballer who plays as a forward for and captains both Saudi Pro League club Al Nassr and the Portugal national team. Widely regarded as one of the greatest players of all time, Ronaldo has won five Ballon d'Or awards,[note 3] a record three UEFA Men's Player of the Year Awards, and four European Golden Shoes, the most by a European player. He has won 33 trophies in his career, including seven league titles, five UEFA Champions Leagues, the UEFA European Championship and the UEFA Nations League. Ronaldo holds the records for most appearances (183), goals (140) and assists (42) in the Champions League, goals in the European Championship (14), international goals (128) and international appearances (205). He is one of the few players to have made over 1,200 professional career appearances, the most by an outfield player, and has scored over 850 official senior career goals for club and country, making him the top goalscorer of all time.
"""

labels = ["person", "award", "date", "competitions", "teams"]

entities = model.predict_entities(text, labels)

for entity in entities:
    print(entity["text"], "=>", entity["label"])
```

```
- Cristiano Ronaldo dos Santos Aveiro => person
- 5 February 1985 => date
- Al Nassr => teams
- Portugal national team => teams
- Ballon d'Or => award
- UEFA Men's Player of the Year Awards => award
- European Golden Shoes => award
- UEFA Champions Leagues => competitions
- UEFA European Championship => competitions
- UEFA Nations League => competitions
- Champions League => competitions
- European Championship => competitions
```

Pretty awesome! 

## New model cards
Whether your model will be used or not usually depends a lot on whether the model cards are good, so I've created two model cards for you:
* https://huggingface.co/tomaarsen/gliner_base ([raw](https://huggingface.co/tomaarsen/gliner_base/raw/main/README.md) link)
* https://huggingface.co/tomaarsen/gliner_multi ([raw](https://huggingface.co/tomaarsen/gliner_multi/raw/main/README.md) link)

Please feel free to copy these READMEs over to your models, and then I will take mine offline again, as I don't want any credit for your models :) The "Usage" snippets already refer to your models. Also, if you want, then you can copy the full repositories, i.e. with the updated model format, but that's up to you.

## Note
There are some design decisions that you can make based on your preferences. For example, the name of `predict_entities`, the dictionary keys that it outputs (`start`, `end`, `text`, `label`), the saved model format, etc. You can make these decisions as you see fit, I don't want to impose my choices onto your work here.

## Note 2
If you're interested, it might be worthwhile to turn this project into a Python package. You might have to move some of the code into a `gliner` or `src` directory and create a `setup.py` or `pyproject.toml`, but then users can do:
```
pip install gliner
```
&
```
from gliner import GLiNER
```

I think that might be valuable.

- Tom Aarsen

### CODE REVIEW

**Comment by urchade** [comment]
Hi Tom, 

That's great ! Thank you so much for your contribution.

Urchade

**Comment by tomaarsen** [comment]
If you're interested, I've just finished up pull requests for your models also:
1. [gliner_base model card](https://huggingface.co/urchade/gliner_base/discussions/1)
2. [gliner_base file format](https://huggingface.co/urchade/gliner_base/discussions/2)
3. [gliner_multi model card](https://huggingface.co/urchade/gliner_multi/discussions/1)
4. [gliner_multi file format](https://huggingface.co/urchade/gliner_multi/discussions/3)

---

- Tom Aarsen

**Comment by urchade** [comment]
Great, I have merged your pull request. Thanks again!

Urchade
